### PR TITLE
Listen for change to height prop

### PR DESCRIPTION
**Pull request by keller-mark**
This adds an optional `height` prop to the `VitessceGrid` component.

It is slightly misleading because it does not _provide_ or _change_ the height of the grid, it simply passes a height value from an outer state, and when `ResponsiveHeightGridLayout` detects a change it will check for new dimensions by calling the `WidthProvider` onWindowResize function:

```sh
    componentDidUpdate(prevProps) {
        if (this.props.height !== prevProps.height) {
            this.onWindowResize();
        }
    }
```

This avoids the need to dispatch a superfluous window.resize event to tell the ResponsiveHeightGridLayout to check its dimensions.

For example, in the portal, when changing from a fixed height to a full-window height, we could update the height prop from `600` to `null`, and the ResponsiveHeightGridLayout would know to check for new dimensions, without an explicit event.

I updated the demo to show this functionality

### CODE REVIEW

File: demo/src/renderDemo.js [hunk 1]
```diff
@@ -54,26 +54,41 @@ function getComponent(name) {
...
+  return (
+    <div style={(isExpanded ? { height: `${fixedHeight}px`, width: `${fixedHeight}px` } : {})}>
+      <button onClick={toggleIsExpanded}>Toggle Expanded Grid</button>
+      <VitessceGrid
+        layout={layout}
+        getComponent={getComponent}
+        draggableHandle={`.${handleClass}`}
+        padding={50}
+        height={isExpanded}
+        rowHeight={(isExpanded ? fixedHeight/4 : undefined)} /* If undefined, will fill window height. */
```
Comment (manzt) [inline demo/src/renderDemo.js:72]: why divided by 4?

File: demo/src/renderDemo.js [hunk 2]
```diff
@@ -54,26 +54,41 @@ function getComponent(name) {
   return registry[name];
 }
 
+function Demo() {
+  const fixedHeight = 600;
+  const [isExpanded, setIsExpanded] = useState(true);
+  function toggleIsExpanded() {
+    setIsExpanded(prev => !prev);
+  }
```
Comment (manzt) [inline demo/src/renderDemo.js:62]: Not super important, but I usually use `useReducer` for this pattern.

```suggestion
  const [isExpanded, toggleIsExpanded] = useReducer(v => !v, true);
```

File: demo/src/renderDemo.js (hunk 1 above)
Comment (keller-mark) [reply demo/src/renderDemo.js:72]: there are four "rows" in the layout in `demo/src/layout.json`

File: demo/src/renderDemo.js [hunk 3]
```diff
@@ -54,26 +54,38 @@ function getComponent(name) {
...
+  return (
+    <div style={(isExpanded ? { height: `${fixedHeight}px`, width: `${fixedHeight}px` } : {})}>
+      <button onClick={toggleIsExpanded}>Toggle Expanded Grid</button>
+      <VitessceGrid
+        layout={layout}
+        getComponent={getComponent}
+        draggableHandle={`.${handleClass}`}
+        padding={50}
+        height={isExpanded}
+        rowHeight={(isExpanded ? fixedHeight/4 : undefined)} /* If undefined, will fill window height. */
```
Comment (ilan-gold) [inline demo/src/renderDemo.js:69]: Is the condition here flipped?  It starts expanded, but then the `rowHeight` is defined?

Comment (keller-mark) [reply demo/src/renderDemo.js]: Yes, good catch. Not sure how I mixed that up 😆

**Pull request merged**